### ISSUE TITLE
Deleting large number of files from Queue crashes RT

### ISSUE THREAD

**Issue by chaimav** [Beep6581/RawTherapee#6435]
RT version 5.8-3075-g5502f52f9, Windows 10, i3 8100, 8 GB RAM

> Steps to reproduce crash:
> 1) Add a large number of files to the queue (I don't know what the actual minimum is, but I know it crashes as low as 230 images)
> 2) Open the queue and select all (Ctrl + A on Windows)
> 3) Press the 'delete' button on the keyboard
> 
> Note: Performing the same process with only 100 files in Step 1 works as expected and removes the files from the queue

**EDIT**
Based on discussions below the proper way to induce the crash is
1) Add a large number of files to the queue
2) While these files are still being processed by the queue 
- select all (Ctrl + A on Windows)
- Press the 'delete' button on the keyboard

**Comment by ghost** [comment]
I'm not able to reproduce this.

I started with 100 (mix of RAWs from many different camera brands and types and rather large TIFFs), then 250,  500 and because I have 4x as much memory as you: 1237 images. No problems removing them from the queue (ctrl-a + del).

Some specs:

* RawTherapee version : 5.8-3158-g21f3f4dfd (very latest development + some extra, not-yet-merged, stuff)
* OS                : Linux (kernel 5.9.15-1 x86_64 amd64)
* Distro            : Debian Buster (rel. 10.11)
* CPU              : AMD Ryzen 5 3600 6-Core Processor
* GPU              : GeForce RTX 2060 SUPER 7979 MiB
* Memory           : 32 Gb + 15 Gb Swap

**Comment by chaimav** [comment]
Thanks for testing it. Do you have access to a Widows machine as well? Can you test your latest version and 3075 on it?
I also see that you tested a variety of formats. Mine were all Fuji RAFs from an X-T20, all with the exact same parameters applied (I was batch processing 500+ images from an event).
I was able to repeat the crash without fail multiple times, trying to home in on the actual offending quantity. I gave up before finding the exact number, but those tried over 230 crashed without fail and under worked fine.

**Comment by heckflosse** [comment]
I can reproduce the crash when adding 1246 files to the queue, switching to the queue while the files are added to the queue and select all in queue and hitting the delete-key while the adding is not complete.

**Comment by ghost** [comment]
> Do you have access to a Widows machine as well? 

Nope, I don't.

I just redid the test using a timelapse experiment (added them twice). All from a Nikon D750 and a total of 3726 images. All images have the same edits applied.

* not able to reproduce the _ctrl-a + del_ method,
* not able to reproduce @heckflosse's way of making RT crash.

I tried both 5 times on my Linux box and the queue is emptied without a problem.

Ingo doesn't mention which OS he uses, but this could be Windows specific.

---

EDIT:

$ cat AboutThisBuild.txt 
Version: 5.8-3158-g21f3f4dfd
Branch: dev
Commit: 21f3f4dfd
Commit date: 2022-03-09
Compiler: gcc 8.3.0
Processor: x86_64
System: Linux
Bit depth: 64 bits
Gtkmm: V3.24.0
Lensfun: V0.3.2.0
Build type: release
Build flags: -Wno-maybe-uninitialized -std=c++11 -march=native -Werror=unused-label -Werror=delete-incomplete -fno-math-errno -Wno-attributes -flto -Wall -Wuninitialized -Wcast-qual -Wno-deprecated-declarations -Wno-unused-result -Wunused-macros -fopenmp -Werror=unknown-pragmas -O3 -DNDEBUG -ftree-vectorize
Link flags:  -march=native -flto
OpenMP support: ON
MMAP support: ON
Build OS: Linux 5.9.0-0.bpo.5-amd64 x86_64
Build date: Wed, 09 Mar 2022 11:02:54 +0000 UTC
Build epoch: 1646823774
Build UUID:

**Comment by heckflosse** [comment]
I'm on Win 10

```
$ cat Release/AboutThisBuild.txt
Version: 5.8-3075-g5502f52f9
Branch: dev
Commit: 5502f52f9
Commit date: 2022-02-21
Compiler: gcc 10.3.0
Processor: undefined
System: Windows
Bit depth: 64 bits
Gtkmm: V3.24.5
Lensfun: V0.3.2.0
Build type: Release
Build flags: -Wno-parentheses -std=c++11 -march=native -Werror=unused-label -Werror=delete-incomplete -fno-math-errno -Wno-attributes -Wall -Wuninitialized -Wcast-qual -Wno-deprecated-declarations -Wno-unused-result -Wunused-macros -fopenmp -Werror=unknown-pragmas -O3 -DNDEBUG -ftree-vectorize
Link flags:  -march=native
OpenMP support: ON
MMAP support: ON
Build OS: MINGW64_NT-10.0-19043 3.3.4-341.x86_64 x86_64
Build date: Tue, 15 Mar 2022 12:30:47 +0000 UTC
Build epoch: 1647347447
Build UUID: 278b3db9-2b9e-44d6-aafa-41f24441aa79
```

**Comment by chaimav** [comment]
> I can reproduce the crash when adding 1246 files to the queue, switching to the queue while the files are added to the queue and select all in queue and hitting the delete-key while the adding is not complete.

Good catch! I tried again with all 550 files but waited for the thumbnails to finish populating before deleting and confirmed it didn't crash. The _quantity_ seems to be a red herring, and the issue is that it is still processing the queue.
